ocaml/xapi/quicktest_cbt.ml

```diff
@@ -22,10 +24,10 @@ let start session_id =
     failed test (Printf.sprintf "%s failed: %s" test_name
                    (ExnHelper.string_of_exn error)) in
 
+  (* Define own exception to return error message to the test try/with and fail within that,
+   * as calling 'failed test' in a function called within the test leads to Hashtbl problems *)
   let test_assert ~test op ~msg =
-    try assert op with (Assert_failure _) ->
-      debug test msg;
-      raise (Failure "test assertion failed") in
+    try assert op with (Assert_failure _) -> raise (Test_assertion_failed msg) in
```

Comment:
Actually we could simplify this further to:
```ocaml
if not op then raise (Test_assertion_failed msg)
```
because as far as I know, `assert` just fails for false values.